marshal.go

```diff
@@ -1117,8 +1117,8 @@ func (x *GoSNMP) unmarshalResponse(packet []byte, response *SnmpPacket) error {
 			return fmt.Errorf("error parsing SNMP packet, packet length %d cursor %d", len(packet), cursor)
 		}
 
-		if maxRepetitions, ok := rawMaxRepetitions.(uint32); ok {
-			response.MaxRepetitions = (maxRepetitions & 0x7FFFFFFF)
+		if maxRepetitions, ok := rawMaxRepetitions.(int); ok {
```

Comment:
Should we be explicit about this being `int32` or `int64` here?

Reply:
Yes, I believe this field is defined as a C `int`, (2 or 4 bytes) so a Go `int` (4-8 bytes) should be fine.